test/t8_geometry/t8_geometry_implementations/t8_gtest_geometry_occ.cxx

```diff
@@ -425,23 +428,29 @@ class class_2d_element_linear_occ_surface: public testing::TestWithParam<t8_ecla
 
 TEST_P (class_2d_element_linear_occ_surface, t8_check_2d_element_linear_occ_surface)
 {
-  double vertices_quad[12] = { 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 1.0, 0.0, 1.0, 1.0, 0.0 };
-  double vertices_tri[9] = { 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 1.0, 1.0, 0.0 };
+  /* Saving the corner vertices for the given element class */
+  const int num_vertices = t8_eclass_num_vertices[eclass];
+  double *vertices = T8_ALLOC (double, 3 * num_vertices);
```

Comment:
You can save this allocation here and directly assign the pointer instead of copying over the vertices.

I think you can write it this way. Let me know if the compiler still complains.
```suggestion
  const double *vertices = &(t8_element_corner_ref_coords[eclass][0][0]);
```
This works since the vertices in `t8_element_corner_ref_coords` are in consecutive order already.

Reply:
You are right, this would be way smoother. Sadfully `t8_cmesh_set_tree_vertices` expects `vertices` as double* argument and we pass a const double* now. So i think we have to stick with the copying, right?

Reply:
@jfussbro You either change `t8_cmesh_set_tree_vertices` to accept `const double*` or you remove the `const` from above suggestion. Does the latter work?

Reply:
@jmark The latter option doesn't work, because than `t8_element_corner_ref_coords` cannot be assigned to vertices. The first option worked though :)